src/js/components/TaskFileDownloadComponent.jsx

```diff
@@ -71,16 +75,26 @@ var TaskFileDownloadComponent = React.createClass({
       return;
     }
 
-    DialogActions
-      .alert(`Request failed or timed out for task ID ${request.taskId}`);
-
     MesosStore.removeListener(MesosEvents.REQUEST_TASK_FILES_COMPLETE,
       this.onMesosRequestTaskFilesComplete);
     MesosStore.removeListener(MesosEvents.REQUEST_TASK_FILES_ERROR,
       this.onMesosRequestTaskFilesError);
 
     this.setState({
-      fileIsRequestedByUser: false
+      fileIsRequestedByUser: false,
+      fileRequestFailed: true
+    });
+  },
+
+  handleMouseOver: function () {
+    this.setState({
+      isPopoverVisible: true
```

Comment:
Where else is this used? I think you can remove this state.

Is ```this.forceUpdate()``` that what you want to achieve?